Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -28,253 +29,357 @@
-  template <std::size_t dim>
-  using CovarianceMatrix = Eigen::Matrix<Scalar, dim, dim>;
-  template <std::size_t dim>
-  using CovarianceMatrixMap = Eigen::Map<CovarianceMatrix<dim>>;
-  template <std::size_t dim>
-  using ConstCovarianceMatrixMap = Eigen::Map<const CovarianceMatrix<dim>>;
-  using EffectiveCovarianceMatrix =
-      Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>;
-  using EffectiveCovarianceMatrixMap = Eigen::Map<EffectiveCovarianceMatrix>;
-  using ConstEffectiveCovarianceMatrixMap =
-      Eigen::Map<const EffectiveCovarianceMatrix>;
-
-  using FullParametersVector = Acts::ActsVector<kFullSize>;
-  using FullCovarianceMatrix = Acts::ActsSquareMatrix<kFullSize>;
-
-  using ProjectionMatrix = Eigen::Matrix<Scalar, Eigen::Dynamic, kFullSize>;
-  using ExpansionMatrix = Eigen::Matrix<Scalar, kFullSize, Eigen::Dynamic>;
-
-  /// Construct from source link, subset indices, and measured data.
-  ///
-  /// @tparam parameters_t Input parameters vector type
-  /// @tparam covariance_t Input covariance matrix type
-  /// @param source The link that connects to the underlying detector readout
-  /// @param subspaceIndices Which parameters are measured
-  /// @param params Measured parameters values
-  /// @param cov Measured parameters covariance
-  ///
-  /// @note The indices must be ordered and must describe/match the content
-  ///   of parameters and covariance.
-  template <typename other_indices_t, std::size_t kSize, typename parameters_t,
-            typename covariance_t>
-  VariableSizeMeasurement(
-      Acts::SourceLink source,
-      const std::array<other_indices_t, kSize>& subspaceIndices,
-      const Eigen::MatrixBase<parameters_t>& params,
-      const Eigen::MatrixBase<covariance_t>& cov)
-      : m_source(std::move(source)) {
-    static_assert(kSize == parameters_t::RowsAtCompileTime,
-                  "Parameter size mismatch");
-    static_assert(kSize == covariance_t::RowsAtCompileTime,
-                  "Covariance rows mismatch");
-    static_assert(kSize == covariance_t::ColsAtCompileTime,
-                  "Covariance cols mismatch");
-
-    m_subspaceIndices.resize(subspaceIndices.size());
-    std::transform(subspaceIndices.begin(), subspaceIndices.end(),
-                   m_subspaceIndices.begin(), [](auto index) {
-                     return static_cast<SubspaceIndex>(index);
-                   });
-
-    parameters<kSize>() = params;
-    covariance<kSize>() = cov;
+  template <std::size_t Size>
+  using FixedProxy = FixedMeasurementProxy<Acts::eBoundSize, Size, false>;
+  template <std::size_t Size>
+  using ConstFixedProxy = FixedMeasurementProxy<Acts::eBoundSize, Size, true>;
+  using VariableProxy = VariableMeasurementProxy<Acts::eBoundSize, false>;
+  using ConstVariableProxy = VariableMeasurementProxy<Acts::eBoundSize, true>;
+
+  MeasurementContainer();
+
+  std::size_t size() const;
+
+  void reserve(std::size_t size);
+
+  std::size_t addMeasurement(std::uint8_t size);
```

Comment:
So this works like in the MTJ, where we get an index back, that we then use to edit the measurement, right?
Maybe a two lines for that document what the return type is would be good.

Reply:
This is how we do it in the track EDM and I copied that over. I changed the return type and added some doc